cpp/src/arrow/memory_pool.cc

```diff
@@ -665,6 +665,20 @@ MemoryPool* default_memory_pool() {
 Status jemalloc_set_decay_ms(int ms) {
   return Status::Invalid("jemalloc support is not built");
 }
+
+Status jemalloc_get_stat(const char* name, size_t* out) {
+  return Status::Invalid("jemalloc support is not built");
```

Comment:
nit: why Invalid here and NotImplemented below?

Reply:
Good point, switching all to `Invalid`.

Reply:
TBH, I'd rather switch them all to `NotImplemented`, and reserve `Invalid` for when the parameter given to the function is invalid.